File: math/2745.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		String[] line = sc.nextLine().split(" ");

		int ans = 0;
		int length = line[0].length();
		for (int i = length - 1; 0 <= i; i--) {
			int charNum = line[0].charAt(i);
			
			if('0' <= charNum && charNum <= '9') {
				charNum = charNum - '0';
			} else {
				charNum = charNum - 'A' + 10;
			}
			ans += charNum * Math.pow(Integer.parseInt(line[1]), length -i -1);
		}
		System.out.println(ans);
	}

}

File: 2019상반기에푼것/math/2745.js
function solution(val, b) {
  let ans = 0;
  for (let i = val.length - 1; 0 <= i; i--) {
    let num = val[i].charCodeAt();
    if ('0'.charCodeAt() <= num && num <= '9'.charCodeAt()) {
      num -= '0'.charCodeAt();
    } else {
      num = num - 'A'.charCodeAt() + 10;
    }
    ans += num * Math.pow(b, val.length - i - 1);
  }

  console.log(ans);
}

solution('ZZZZZ', 36);
solution('9876543210', 36);
